lib/facter/util/fact.rb

```diff
@@ -26,13 +26,13 @@ class Facter::Util::Fact
   # @api private
   def initialize(name, options = {})
     @name = name.to_s.downcase.intern
+    @recursion_key = "facter_searching_#{@name}".intern
```

Comment:
Why is this not a boolean to start?

Reply:
Because we use it as a key to a value in the map of current thread vars. We don't change it to a boolean later, but rather the value it points to.

Reply:
oh oh oh wow, reading is hard 🤦‍♀